### ISSUE TITLE
Add validation logic to query.Spec

### ISSUE THREAD

**Issue by adamperlin** [influxdata/platform#259]
From ifql created by [nathanielc](https://github.com/nathanielc) : influxdata/ifql#299  

Currently most of the validation logic is contained within the ifql argument handling and not part of the query OperationSpecs. Since the query Spec is part of the public API we need explicit validation logic at that layer. This is important with the use of transpilers.

**Comment by nathanielc** [comment]
This will be a non issue once https://github.com/influxdata/flux/pull/298 is complete
